lib/dialects/abstract/query-generator.js

```diff
@@ -1135,8 +1136,9 @@ class QueryGenerator {
         return `json_unquote(json_extract(${quotedColumn},${pathStr}))`;
 
       case 'postgres':
+        const join = isJson ? '#>' : '#>>';
```

Comment:
So this would become 

const join = options.jointFragment || "#>>";

Reply:
My worry with this option is it won't solve the issue because the fragment is based on data type and you can have a where clause that tries to use both and it will fail because you can only use the default fragment or the option.  Example

Data Structure
```js
const userData = {
  id: DataTypes.STRING,
  name: DataTypes.STRING,
  profile: DataTypes.JSONB,
};
```

Demo Profile
```js
const profile = {
  lastLogin: 2021-12-19T21:26:58.489Z,
  failedLoginAttempts: 2,
  locked: false,
  settings: [
    { name: 'allowEmails', value: true },
    { name: 'allowPasswordReset', value: true },
  ],
};
```

If we want to find out when a user last logged in and if they can reset their password you would need a query similar to this (I'm writing this off the cuff so syntax may not be perfect):

`SELECT * FROM userData WHERE id = '1a2b3c4d' AND profile #>> lastLogin > ${now() + 1 hour} AND profile #> settings @> '[{ name: 'allowPasswordReset', value: true }]';`

If we passed in options.joinFragment = '#>' that query would be:
`SELECT * FROM userData WHERE id = '1a2b3c4d' AND profile #> lastLogin > ${now() + 1 hour} AND profile #> settings @> '[{ name: 'allowPasswordReset', value: true }]';`
because there is no way the query can tell which of the two needs to be replaced.

Reply:
Agreed